src/adapters/leasing-adapter.ts

```diff
@@ -562,6 +562,24 @@ const getOffersByListingId = async (
   }
 }
 
+const getActiveOfferByListingId = async (
+  listingId: number
+): Promise<AdapterResult<Offer, 'not-found' | 'unknown'>> => {
+  try {
+    const res = await axios(
+      `${tenantsLeasesServiceUrl}/offers/listing-id/${listingId}/active`
+    )
+
+    if (!res.data.content) {
+      return { ok: false, err: 'not-found' }
```

Comment:
Can we add ` statusCode: number ` to the AdapterResult type? So that we can simplify the error handling the endpoints using this adapter function, like this: 

```
if(!result.ok){
    ctx.status = result.statusCode
    ctx.body = { error: result.err, ...metadata }
    return
}    
```

Reply:
I think we should either add another type or add statuscode as optional as there are many cases where we don't want to care about status code

Reply:
Sure, it could be optional. The gist here is to simplify the calling code

Reply:
https://github.com/Bostads-AB-Mimer/onecore-core/pull/188/commits/9e730901cab3f26b039463eb26e390f49156d460

I tried it here. It gets rather ugly IMO.. :D

Another thing is that now when we just proxy whatever leasing responds with, the contract (especially obvious in regards to the swagger docs) becomes less obvious since leasing could potentially change its response codes and core wouldn't know about it. What do you think?